File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/EventChatV2.java
package Auxiliary_Modules;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;

public class EventChatV2 {
    private final Map<String, LimUniqueResponder> dic = new HashMap<>();
    private final HashSet<String> modifiedKeys = new HashSet<>();
    private final int lim;

    // Constructor
    public EventChatV2(int lim) {
        this.lim = lim;
    }

    public HashSet<String> getModifiedKeys() {
        return modifiedKeys;
    }
    public Boolean keyExists(String key){
        // if the key was active true is returned
        return modifiedKeys.contains(key);
    }

    // Add items
    public void addItems(LimUniqueResponder ur, String... args) {
        for (String arg : args) {
            dic.put(arg, ur.clone());
        }
    }


    public void addFromDB(String key, String value){
        if (value.isEmpty()||value.equals("null")){return;}
        AXStringSplit tool1 = new AXStringSplit();
        String[] values = tool1.split(value);
        if (!dic.containsKey(key)){dic.put(key, new LimUniqueResponder(lim));}
        for (String item : values) {
            dic.get(key).addResponse(item);
        }
    }
    // Add key-value pair
    public void addKeyValue(String key, String value) {
        modifiedKeys.add(key);
        if (dic.containsKey(key)) {
            dic.get(key).addResponse(value);
        } else {
            dic.put(key, new LimUniqueResponder(lim));
            dic.get(key).addResponse(value);
        }
    }
    public void addKeyValues(ArrayList<AXKeyValuePair> elizaResults) {
        for (AXKeyValuePair pair : elizaResults) {
            // Access the key and value of each AXKeyValuePair object
            addKeyValue(pair.getKey(),pair.getValue());
        }
    }
    // Get response
    public String response(String in1) {
        return dic.containsKey(in1) ? dic.get(in1).getAResponse() : "";
    }
    public String responseLatest(String in1) {
        return dic.containsKey(in1) ? dic.get(in1).getLastItem() : "";
    }
    public String getSaveStr(String key) {
        return dic.get(key).getSavableStr();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/EventChatV2.kt
package auxiliary_modules

class EventChatV2    // Constructor
    (private val lim: Int) {
    private val dic: MutableMap<String, LimUniqueResponder> = HashMap()
    val modifiedKeys = HashSet<String>()

    fun keyExists(key: String): Boolean {
        // if the key was active true is returned
        return modifiedKeys.contains(key)
    }

    // Add items
    fun addItems(ur: LimUniqueResponder, vararg args: String) {
        for (arg in args) {
            dic[arg] = ur.clone()
        }
    }

    fun addFromDB(key: String, value: String) {
        if (value.isEmpty() || value == "null") {
            return
        }
        val tool1 = AXStringSplit()
        val values = tool1.split(value)
        if (!dic.containsKey(key)) {
            dic[key] = LimUniqueResponder(lim)
        }
        for (item in values) {
            dic[key]!!.addResponse(item)
        }
    }

    // Add key-value pair
    fun addKeyValue(key: String, value: String?) {
        modifiedKeys.add(key)
        if (dic.containsKey(key)) {
            dic[key]!!.addResponse(value!!)
        } else {
            dic[key] = LimUniqueResponder(lim)
            dic[key]!!.addResponse(value!!)
        }
    }

    fun addKeyValues(elizaResults: ArrayList<AXKeyValuePair>) {
        for (pair in elizaResults) {
            // Access the key and value of each AXKeyValuePair object
            addKeyValue(pair.key, pair.value)
        }
    }

    // Get response
    fun response(in1: String): String {
        return if (dic.containsKey(in1)) dic[in1]!!.aResponse else ""
    }

    fun responseLatest(in1: String): String {
        return if (dic.containsKey(in1)) dic[in1]!!.lastItem else ""
    }

    fun getSaveStr(key: String): String {
        return dic[key]!!.savableStr
    }
}